### DISCUSSION TITLE
Proposed: remove HTTPS_SERVER define

### DISCUSSION THREAD

**Discussion by drbyte** [zencart/zencart#5794] [Ideas]
Since modern browsers expect SSL all the time, there's no need to support intermingled http/https like we did in original Zen Cart back when SSL was only used by those who were security-minded.

Thus the double set of `HTTP_SERVER` + `DIR_WS_CATALOG` and `HTTPS_SERVER` + `DIR_WS_HTTPS_CATALOG` is redundant.

I'd like to remove the `HTTPS_SERVER` and `DIR_WS_HTTPS_CATALOG` defines and all the internal infrastructure that refers to them. Both Catalog and Admin.

Caveat: there may be plugins that have hard-coded things to reference these defines. 
But chances are those plugins need other updates anyway so I'd be satisfied with simply documenting the change.

**Comment by scottcwilson** [comment]
Can we just define the things which are removed in core the way we did with other variables we removed from the configure file?

**Comment by proseLA** [comment]
i'm a fan of the removal.  either you keep up or you get left behind.  i am a fan of less configuration vars.  plugins either get updated or left behind.

**Comment by dbltoe** [comment]
I'm one of the first to promote keeping current but, what about the showcase sites?  Is it the browsers that are going to be the bad guys in this or is it going to be ZC not working without SSL?  Just would like to be able to blame the "other guy".

**Comment by drbyte** [comment]
> what about the showcase sites?

I'm not sure I follow whatever you meant to say here.

**Comment by dbltoe** [comment]
Searching for "Welcome, please enjoy our online showcase." comes up with some 1600 returns.  About a hundred are how to change it so, around 1500 are not selling and have not changed the default.  If you figure that just a fourth of the actual non-selling sites did change the greeting, there would be about 2000 sites that could operate fine without an SSL.  Some have an SSL regardless of neither selling nor logging folks in but, most that I looked at did not have a link to log in.  Since the lack of SSL would probably not be triggered in this case.  Is it enough of an occurrence to warrant any consideration?

**Comment by drbyte** [comment]
@dbltoe I think you're saying "there are some sites that do operate entirely without SSL".

Yes I agree. Even in a localhost test environment some will not use SSL. That's simple: just set an http URL into HTTP_SERVER.

But when a site is live and has a functional SSL certificate, then HTTP_SERVER would get defined to an `https://` URL, and the store would always generate https URLs when any URL-generating functions are called, such as `zen_href_link`.

My proposal is to remove a pointless duplicitous set of defines that not only create confusion for the end-user but also lead to unnecessary code complexity.

**Comment by drbyte** [comment]
The old set of defines existed primarily to allow the use of "Shared SSL Certificates", which is a largely dead concept today.
Years ago often SSL was only available by sharing an SSL cert from the hosting company which then path-mapped into the user's home directory to serve the SSL-only pages. 
The URIs for secured pages were like `https://hosting_co.com/username/domain/index.php?main_page=login`
This approach required that HTTPS_SERVER be set to the hosting company's domain, and DIR_WS_HTTPS_CATALOG  be set to the extra subdirectories path.

All that is no longer needed today because basically nobody offers "shared SSL" anymore since SSL certificates are free.

Further, it's a bad idea to use shared SSL nowadays because nobody wants search engines indexing their hosting company's domain name when crawling their store.
In fact, that's why the `$request_type` variable was introduced and the `SSL`/`NONSSL` parameter to `zen_href_link` along with the `ENABLE_SSL` constant. These were present to "only use SSL on pages that require a logged-in-session":
 ... to cause search engines to essentially bypass/ignore these long Shared SSL URIs
 ... and to avoid confusing the shopper who visited `http://example.com` but got redirected to `https://hosting_co.com/username/domain/` instead.

And CORS and other cross-domain security rules nowadays would make javascript and ajax and cookies non-functioning if shared-SSL were used.

So, ya, retiring ENABLE_SSL is also part of this proposal. 
Can't remove the SSL/NONSSL flag from zen_href_link because it's in the middle of a bunch of other parameters in the function signature. We'd just remove the logic from the function that does anything with that value, removing complexity.

**Comment by scottcwilson** [comment]
> The old set of defines existed primarily to allow the use of "Shared SSL Certificates", which is a largely dead concept today.

Yep - it was removed from the new docs for this reason.

**Comment by dbltoe** [comment]
Back in school, I'd tell the coach, "I lost it in the lights."  Today, I lose it in the lines on the glasses.

👍
